src/Symfony/Component/Cache/Simple/Psr6Cache.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Simple;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheException as Psr6CacheException;
+use Psr\SimpleCache\CacheInterface;
+use Psr\SimpleCache\CacheException as SimpleCacheException;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class Psr6Cache implements CacheInterface
+{
+    private $pool;
+    private $createCacheItem;
+
+    public function __construct(CacheItemPoolInterface $pool)
+    {
+        $this->pool = $pool;
+
+        if ($pool instanceof Adapter\AdapterInterface) {
+            $this->createCacheItem = \Closure::bind(
+                function ($key, $value) {
+                    CacheItem::validateKey($key);
+                    $item = new CacheItem();
+                    $item->key = $key;
+                    $item->value = $value;
+
+                    return $item;
+                },
+                null,
+                CacheItem::class
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($key, $default = null)
+    {
+        try {
+            $item = $this->pool->getItem($key);
+        } catch (SimpleCacheException $e) {
+            throw $e;
+        } catch (Psr6CacheException $e) {
+            throw new InvalidArgumentException($e->getMessage());
```

Comment:
there could be a second reason: misbehaving cache implementation. In this case, knowing the original location of the exception could help understand whether the issue is in the key, or in the key validation. And btw, it would still point people to the code implementing the key validation, showing them the logic used to validate it.
So IMO, it is still a good idea to link the original exception.